apis/Google.Devtools.AspNet/Google.Devtools.AspNet.IntegrationTests/CloudTrace/CloudTraceIntregrationTest.cs

```diff
@@ -0,0 +1,288 @@
+        {
+            TraceIdFactory traceIdFactory = TraceIdFactory.Create();
+            Trace trace = new Trace()
+            {
+                ProjectId = ProjectId,
+                TraceId = traceIdFactory.NextId()
+            };
+            return SimpleManagedTracer.Create(consumer, trace, null);
+        }
+
+        /// <summary>
+        /// Gets a trace that contains a span with the given name.
+        /// </summary>
+        private async Task<Trace> GetTrace(string spanName)
+        {
+            int sleepTime = 0;
+            while (sleepTime < TimeoutMilliseconds)
+            {
+                sleepTime += SleepIntervalMilliseconds;
+                Thread.Sleep(SleepIntervalMilliseconds);
+
+                ListTracesRequest request = new ListTracesRequest
+                {
+                    ProjectId = ProjectId,
+                    StartTime = StartTime,
+                    View = ListTracesRequest.Types.ViewType.Complete
+                };
+                ListTracesResponse response = await GrpcClient.ListTracesAsync(request);
+                List<Trace> traces = response.Traces.Where(t => t.Spans.Any(s => s.Name.Equals(spanName))).ToList();
+                if (traces.Count != 0)
+                {
+                    return traces[0];
+                }
+            }
+            return null;
+        }
+
+        private string CreateRootSpanName(string name) => $"{TestId}-{name}";
+
+        [Fact]
+        public async Task Trace_Simple()
+        {
+            string rootSpanName = CreateRootSpanName(nameof(Trace_Simple));
+            GrpcTraceConsumer consumer = CreateGrpcTraceConsumer();
+            SimpleManagedTracer tracer = CreateSimpleManagedTracer(consumer);
+
+            tracer.StartSpan(rootSpanName);
+            Thread.Sleep(10);
+            tracer.EndSpan();
+
+            Trace trace = await GetTrace(rootSpanName);
+            Assert.NotNull(trace);
+            Assert.Single(trace.Spans);
+        }
+
+        [Fact]
+        public async Task Trace_Simple_Buffer()
+        {
+            string rootSpanName = CreateRootSpanName(nameof(Trace_Simple_Buffer));
+            BufferingTraceConsumer consumer = BufferingTraceConsumer.Create(CreateGrpcTraceConsumer());
+            SimpleManagedTracer tracer = CreateSimpleManagedTracer(consumer);
+
+            // Create annotations with very large labels to ensure the buffer is flushed.
+            string label = "1234567890";
+            while (label.Length < 20000)
+            {
+                label += label;
+            }
+            Dictionary<string, string> annotation = new Dictionary<string, string>
+            {
+                { "key-one", label },
+                { "key-two", label },
+                { "key-three", label },
+                { "key-four", label },
+                { "key-five", label },
+            };
+
+            tracer.StartSpan(rootSpanName);
+            Thread.Sleep(10);
```

Comment:
If this is meant to make sure the clock has ticked over, we should probably use 15 instead of 10, as that's the Windows coarse clock granularity, IIRC. But let's make it a static readonly field so we only need to update one place.

Alternatively, we could write a method to "wait until the clock ticks over" which would repeatedly sleep until it saw a different result - that may well be a better plan.